packages/snowpack/src/types/snowpack.ts

```diff
@@ -72,6 +72,8 @@ export interface SnowpackPlugin {
   run?(options: PluginRunOptions): Promise<unknown>;
   /** optimize the entire built application */
   optimize?(options: PluginOptimizeOptions): Promise<void>;
+  /** cleanup any long-running instances/services before exiting.  */
+  cleanup?(): void | Promise<void>;
```

Comment:
Like we talked about, I think this is a good approach. From the plugin’s perspective, it doesn’t know how many times `load()` or `transform()` will be called. Nor will Snowpack, the way it’s working now.

I think this solution is better than trying to give context to the other functions whether or not it’s the last run (but also, I don’t like giving plugin hooks the ability to build differently based on the state of the build… that seems like it could backfire)